codalab/worker/main.py

```diff
@@ -147,44 +148,48 @@ def connect_to_codalab_server(server, password_file):
 
 def main():
     args = parse_args()
+    # This quits if connection unsuccessful
     bundle_service = connect_to_codalab_server(args.server, args.password_file)
+    # Configure logging
     logging.basicConfig(
         format='%(asctime)s %(message)s', level=(logging.DEBUG if args.verbose else logging.INFO)
     )
-
-    max_work_dir_size_bytes = parse_size(args.max_work_dir_size)
-    max_image_cache_size_bytes = (
-        parse_size(args.max_image_cache_size) if args.max_image_cache_size else None
-    )
-
+    if args.shared_file_system:
+        # No need to store bundles locally if filesystems are shared
+        local_bundles_dir = None
+        # Also no need to download dependencies if they're on the filesystem already
+        dependency_manager = None
+    else:
+        local_bundles_dir = os.path.join(args.work_dir, 'runs')
+        dependency_manager = DependencyManager(
+            os.path.join(args.work_dir, 'dependencies-state.json'),
+            bundle_service,
+            args.work_dir,
+            args.max_work_dir_size,
+        )
+    # Set up local directories
     if not os.path.exists(args.work_dir):
         logging.debug('Work dir %s doesn\'t exist, creating.', args.work_dir)
         os.makedirs(args.work_dir, 0o770)
+    if local_bundles_dir and not os.path.exists(local_bundles_dir):
```

Comment:
Create both local dirs here for consistency (before, the latter was created within the constructor of the Worker class)